tests/src/kernel/hipLaunchParm.cpp

```diff
@@ -23,75 +23,768 @@ THE SOFTWARE.
+                    dim3(1), 0, 0, hipLaunchKernelStruct_h4,
+                    result_d);
+    ResultValidation();
+
+    // Test: Passing struct with pointer object to a hipLaunchKernelGGL()
+    hipLaunchKernelStruct_t5 hipLaunchKernelStruct_h5;
+    char* cp_d5;  // This is passed as pointer to struct member
+    // allocating memory for char pointer on device
+    HIPCHECK(hipMalloc((void**)&cp_d5, sizeof(char)));
+    HIPCHECK(hipMemset(cp_d5, 'p', sizeof(char)));
+    hipLaunchKernelStruct_h5.c1 = 'c';
+    hipLaunchKernelStruct_h5.cp = cp_d5;
+    hipLaunchKernelGGL(HIP_KERNEL_NAME(hipLaunchKernelStructFunc5),
+                    dim3(BLOCK_DIM_SIZE),
+                    dim3(1), 0, 0, hipLaunchKernelStruct_h5,
+                    result_d);
+    ResultValidation();
+
+    // Test: Passing struct with aligned(8)
+    hipLaunchKernelStruct_t6 hipLaunchKernelStruct_h6;
+    hipLaunchKernelStruct_h6.c1 = 'c';
+    hipLaunchKernelStruct_h6.si = 1;
+    hipLaunchKernelGGL(HIP_KERNEL_NAME(hipLaunchKernelStructFunc6),
+                    dim3(BLOCK_DIM_SIZE),
+                    dim3(1), 0, 0, hipLaunchKernelStruct_h6,
+                    result_d);
+    // alignment is broken hence disabled the validation part
+    #if ENABLE_ALIGNMENT_TEST
+    ResultValidation();
+    #endif
+
+
+    // Test: Passing struct with aligned(16)
+    hipLaunchKernelStruct_t7 hipLaunchKernelStruct_h7;
+    hipLaunchKernelStruct_h7.c1 = 'c';
+    hipLaunchKernelStruct_h7.si = 1;
+    #if ENABLE_ALIGNMENT_TEST  // This is broken on small bar
+    hipLaunchKernelGGL(HIP_KERNEL_NAME(hipLaunchKernelStructFunc7),
+                    dim3(BLOCK_DIM_SIZE),
+                    dim3(1), 0, 0, hipLaunchKernelStruct_h7,
+                    result_d);
+    ResultValidation();
+    #endif
+
+    // Test: Passing struct with packed aligned to 6Bytes
+    hipLaunchKernelStruct_t8 hipLaunchKernelStruct_h8;
+    hipLaunchKernelStruct_h8.c1 = 'c';
+    hipLaunchKernelStruct_h8.si = 1;
+    hipLaunchKernelGGL(HIP_KERNEL_NAME(hipLaunchKernelStructFunc8),
+                    dim3(BLOCK_DIM_SIZE),
+                    dim3(1), 0, 0, hipLaunchKernelStruct_h8,
+                    result_d);
+    // alignment is broken hence disabled the validation part
+    #if ENABLE_ALIGNMENT_TEST
+    ResultValidation();
+    #endif
+
+
+    // Test: Passing const struct object to a hipLaunchKernelGGL()
+    uint32_t* ip_d9;
+    // allocating memory for char pointer on device
+    HIPCHECK(hipMalloc((void**)&ip_d9, sizeof(uint32_t)));
+    HIPCHECK(hipMemset(ip_d9, 1, sizeof(uint32_t)));
+    // ip_d9 passed as pointer to struct member, struct.ip = &ip_d9
+    const hipLaunchKernelStruct_t9 hipLaunchKernelStruct_h9 = {'c', ip_d9};
+    hipLaunchKernelGGL(HIP_KERNEL_NAME(hipLaunchKernelStructFunc9),
+                    dim3(BLOCK_DIM_SIZE),
+                    dim3(1), 0, 0, hipLaunchKernelStruct_h9,
+                    result_d);
+    #if ENABLE_DECLARE_INITIALIZATION_POINTER
```

Comment:
does this test fail with hip-clang? This is unexpected.

Reply:
Yes Sam, this is not failing.